src/librustdoc/html/markdown.rs

```diff
@@ -440,6 +502,23 @@ pub fn render(w: &mut fmt::Formatter,
                 Event::Start(Tag::Strong) => {
                     strong(parser, buffer, toc_builder, shorter, id);
                 }
+                Event::Start(Tag::Rule) => {
+                    rule(parser, buffer, toc_builder, shorter, id);
+                }
+                Event::Start(Tag::FootnoteDefinition(ref def)) => {
+                    let mut content = String::new();
+                    footnote(parser, &mut content, toc_builder, shorter, def.as_ref().to_owned(),
+                             id);
+                    let cur_len = parser.footnotes.len() + 1;
+                    parser.footnotes.push(format!("<li id=\"ref{}\">{}<a href=\"#supref{0}\" \
+                                                   rev=\"footnote\">↩</a></li>",
+                                                  cur_len, content));
+                }
+                Event::FootnoteReference(_) => {
+                    buffer.push_str(&format!("<sup id=\"supref{0}\"><a href=\"#ref{0}\">{0}</a>\
+                                              </sup>",
+                                             parser.get_next_footnote_id()));
```

Comment:
Footnote ids cannot be this simple either. The same footnote can be referenced from multiple places. So you can't simply keep incrementing the id whenever a footnote reference is found. You'd need to use a map.